plugins/by-name/lsp-signature/default.nix

```diff
@@ -0,0 +1,208 @@
+          - Return `false` to disable.
+          - Return `'right_align'` to display hint right aligned in the current line.
+        '';
+
+    hi_parameter = defaultNullOpts.mkStr "LspSignatureActiveParameter" ''
+      How your parameter will be highlighted.
+    '';
+
+    handler_opts =
+      defaultNullOpts.mkAttributeSet
+        {
+          border = "rounded";
+        }
+        ''
+          `border` can be double, rounded, single, shadow, none or a lua table of borders.
+        '';
+
+    always_trigger = defaultNullOpts.mkBool false ''
+      Sometime show signature on new line or in middle of parameter can be confusing.
+    '';
+
+    auto_close_after = defaultNullOpts.mkLua "nil" ''
+      Autoclose signature float win after x seconds, disabled if `nil`.
+    '';
+
+    extra_trigger_chars = defaultNullOpts.mkListOf lib.types.str [ ] ''
+      Array of of extra characters that will trigger signature completion.
+    '';
+
+    zindex = defaultNullOpts.mkInt 200 ''
+      By default it will be on top of all floating windows, set to <= 50 to send it to the bottom.
+    '';
+
+    padding = defaultNullOpts.mkStr "" ''
+      Character to pad on left and right of signature.
+    '';
+
+    transparency = defaultNullOpts.mkStrLuaOr (lib.types.ints.between 1 100) "nil" ''
+      Disabled by default, allow floating window transparent value 1~100.
+    '';
+
+    shadow_blend = defaultNullOpts.mkInt 36 ''
+      If you're using shadow as border, use this to set the opacity.
+    '';
+
+    shadow_guibg = defaultNullOpts.mkStr "Green" ''
+      If you're using shadow as border, use this to set the color.
+    '';
+
+    time_interval = defaultNullOpts.mkInt 200 ''
+      Timer check interval. Set to a lower value if you want to reduce latency.
+    '';
+
+    toggle_key = defaultNullOpts.mkLua "nil" ''
+      Toggle signature on and off in insert mode.
+    '';
+
+    toggle_flip_floatwin_setting = defaultNullOpts.mkStrLuaOr lib.types.bool false ''
+      - `true`: toggle.
+      - `floating_windows`: true|false setting after toggle key pressed.
+      - `false`: floating_windows setup will not change, toggle_key will pop up signature helper, but signature.
+
+      May not popup when typing, depending on floating_window setting.
+    '';
+
+    select_signature_key = defaultNullOpts.mkLua "nil" ''
+      The keybind for cycling to next signature.
+    '';
+
+    move_cursor_key = defaultNullOpts.mkLua "nil" ''
+      imap, use nvim_set_current_win to move cursor between current window and floating window. Once moved to floating window, you can use <M-d>, <M-u> to move cursor up and down.
+    '';
+
+    # TODO: figure out the types
+    keymaps = defaultNullOpts.mkListOf lib.types.anything [ ] ''
+      related to move_cursor_key. the keymaps inside floating window with arguements of bufnr. 
+      It can be a function that sets keymaps.
+      <M-d> and <M-u> are default keymaps for moving the cursor up and down.
+    '';
```

Comment:
```suggestion
    move_cursor_key = defaultNullOpts.mkLua "nil" ''
      imap, use `nvim_set_current_win` to move cursor between current window and floating window.
      Once moved to floating window, you can use `<M-d>`, `<M-u>` to move cursor up and down.
    '';

    # TODO: figure out the types
    keymaps = defaultNullOpts.mkListOf lib.types.anything [ ] ''
      Related to `move_cursor_key`.
      The keymaps inside floating window with arguments of `bufnr`.
      It can be a function that sets keymaps.
      `<M-d>` and `<M-u>` are default keymaps for moving the cursor up and down.
    '';
```

Reply:
- Try to wrap "snippets" in backquotes: `foo`.
- Ensure that all sentences start with a capital letter.
- Usually, we break line after each period (to avoid too-long lines). This does not impact the final render.

Reply:
i think i fixed everything mentioned